File: boj_9733.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedHashMap;
import java.util.StringTokenizer;

public class Main {

    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        LinkedHashMap<String, Integer> map = new LinkedHashMap<>();
        map.put("Re", 0);
        map.put("Pt", 1);
        map.put("Cc", 2);
        map.put("Ea", 3);
        map.put("Tb", 4);
        map.put("Cm", 5);
        map.put("Ex", 6);
        int[] cnt = new int[7];
        int total = 0;
        String str;
        while ((str = br.readLine()) != null) {
            StringTokenizer st = new StringTokenizer(str);
            while (st.hasMoreTokens()) {
                total++;
                String behave = st.nextToken();
                if (!map.containsKey(behave)) continue;
                cnt[map.get(behave)]++;
            }
        }
        for (String key : map.keySet()) {
            System.out.printf("%s %d %.2f\n", key, cnt[map.get(key)], (float) cnt[map.get(key)] / total);
        }
        System.out.println("Total " + total + " 1.00");
    }
}

File: boj_9733.cpp
#include <iostream>
#include <map>

using namespace std;

int main() {
    map<string, int> map;
    map.insert(make_pair("Re", 0));
    map.insert(make_pair("Pt", 1));
    map.insert(make_pair("Cc", 2));
    map.insert(make_pair("Ea", 3));
    map.insert(make_pair("Tb", 4));
    map.insert(make_pair("Cm", 5));
    map.insert(make_pair("Ex", 6));

    int cnt[7] = {
        0,
    };
    int total;

    string behave;
    while (!(cin >> behave).eof()) {
        total++;
        if (!map.count(behave)) continue;
        cnt[map[behave]]++;
    }
    printf("Re %d %.2f\n", cnt[map["Re"]], ((float)cnt[map["Re"]] / total));
    printf("Pt %d %.2f\n", cnt[map["Pt"]], ((float)cnt[map["Pt"]] / total));
    printf("Cc %d %.2f\n", cnt[map["Cc"]], ((float)cnt[map["Cc"]] / total));
    printf("Ea %d %.2f\n", cnt[map["Ea"]], ((float)cnt[map["Ea"]] / total));
    printf("Tb %d %.2f\n", cnt[map["Tb"]], ((float)cnt[map["Tb"]] / total));
    printf("Cm %d %.2f\n", cnt[map["Cm"]], ((float)cnt[map["Cm"]] / total));
    printf("Ex %d %.2f\n", cnt[map["Ex"]], ((float)cnt[map["Ex"]] / total));
    printf("Total %d 1.00\n\n", total);
    return 0;
}
